File: frontend/ATS/src/app/archive/models/archived-entity.ts
import { EntityType } from 'src/app/shared/enums/entity-type.enum';
import { User } from 'src/app/users/models/user';

export interface ArchivedEntity {
  id: string;
  entityId: string;
  entityType: EntityType;
  expirationDate: Date;
  userId: string;
  user: User;
}

File: backend/src/Domain/Entities/ArchivedEntity.cs
using Domain.Common;
using Domain.Enums;
using System;

namespace Domain.Entities
{
    public class ArchivedEntity: Entity
    {
        public string EntityId { get; set; }
        public EntityType EntityType { get; set; }
        public DateTime ExpirationDate { get; set; }
        public string UserId { get; set; }
        public User User { get; set; }
    }
}
